server/src/test/java/org/opensearch/index/engine/InternalEngineTests.java

```diff
@@ -1832,14 +1832,14 @@ public void testForceMergeWithSoftDeletesRetentionAndRecoverySource() throws Exc
                 }
             }
             for (int i = 0; i < numDocs; i++) {
-                boolean useRecoverySource = randomBoolean() || omitSourceAllTheTime;
                 ParsedDocument doc = testParsedDocument(Integer.toString(i), null, testDocument(), B_1, null, useRecoverySource);
-                if (randomBoolean()) {
+                boolean isDeleted = randomBoolean();
+                if (isDeleted) {
```

Comment:
Does this mean that if `isDeleted` is true, for each document, we delete it firstly, and then add them back? If so the randomness is also lost.